### PR TITLE
Feature sockets setup

### PR DESCRIPTION
**Pull request by rudream**
Set up sockets with authentication. Had to create a new auth middleware because it wasn't possible to integrate Val's authMiddleware.js into socket.io

### CODE REVIEW

File: server/socketAPI.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const socket_io = require("socket.io");
```
Comment (dproc96) [inline server/socketAPI.js]: This isn't in the package.json, can you npm install it? Also this variable should be updated to camelCase

**Review by dproc96 — COMMENTED** [review]
The socket is not connecting on my end, if it's working for you, any ideas?

File: server/socketAPI.js (hunk 1 above)
Comment (rudream) [reply server/socketAPI.js]: Yes, I'm not sure why it didn't get included in the package.json when I npm installed it. I think I may have accidentally overwritten it during merging.

**Comment by rudream** [comment]
> The socket is not connecting on my end, if it's working for you, any ideas?

It works fine on my end. I was having a similar issue earlier as well and restarting my computer and running the server again fixed it.

**Review by MinaKhamesi — APPROVED** [review]
@rudream good job Yassine. It worked for me.

**Review by Vaibhav-Kambli — APPROVED** [review]
Looks great @rudream, sockets connected with proper user details!

**Pull request merged**